src/core/opamVersionCompare.ml

```diff
@@ -80,8 +63,8 @@ let compare_chunks x y =
   and yl = String.length y
   in
   let rec loop_lexical xi yi =
-    assert (xi <= xl && yi <= yl);
-    match (xi=xl,yi=yl) with (* which of x and y is exhausted? *)
+    assert (Int.compare xi xl <= 0 && Int.compare yi yl <= 0);
```

Comment:
For readability, i'm wondering if it is better to have the annotation form:
 `(xi <= (xl:int) && yi <= (yl:int))` vs `(Int.compare xi xl <= 0 && Int.compare yi yl <= 0)`?

Reply:
done